app/models/site_setting.rb

```diff
@@ -143,6 +143,12 @@ def self.allowed_unicode_username_characters_regex
     end
   end
 
+  def self.group_setting_map(setting)
```

Comment:
A follow-up PR can be done to replace other places in the codebase that do this splitting pattern. Alternatively, I did think of getting all the `group_list` settings and dynamically creating methods for them to do this split/map e.g. `personal_message_enabled_groups_map`, so we can do `SiteSetting.personal_message_enabled_groups_map`